src/NHibernate.Validator/Engine/ClassValidator.cs

```diff
@@ -565,6 +566,175 @@ public IEnumerable<InvalidValue> GetPotentialInvalidValues(string propertyName,
 			return GetPotentialInvalidValues(propertyName, value, null);
 		}
 
+
+		/// <summary>
+		/// Create validators based on hibernate metadata if appropriative validator not defined explicitly
+		/// </summary>
+		/// <param name="properties">Hibernate metadata to analize</param>
+		public void ConfigureFrom(IEnumerable<Property> properties)
+		{
+			foreach(var prop in properties)
+			{
+
+				var propTp = prop.Type.ReturnedClass;
+				var propInfo = entityType.GetProperty(prop.Name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance); 
+
+				if (!prop.IsNullable 
+					&& (!propTp.IsValueType
+						 || (propInfo != null && propInfo.PropertyType.IsGenericType && propInfo.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>))))
+				{
+					ConfigurePropValidatotBasedOnMapping<NotNullAttribute>(prop, null);
+				}
+
+				if (propTp== typeof(string))
+				{
+					ConfigurePropValidatotBasedOnMapping<LengthAttribute>(prop
+						, (attr, col) => attr.Max = col.Length);
+				}
+				else if (propTp.IsEnum)
+				{
+					ConfigurePropValidatotBasedOnMapping<EnumAttribute>(prop, null);
+				}
+				else if (   propTp == typeof(decimal) 
+						 || propTp == typeof(double)
+						 || propTp == typeof(float)

```

Comment:
The normal semantic for `double` and `float` is to be ... floating. Meaning they can have a very varying number of fractional digits in the same column, without this causing issues.
Moreover, the `Column` type has a quite nasty behavior in my opinion, defaulting scale to two when not defined and precision to 19. In NHibernate code base, this behavior is not used, this is "just an old trash" which should be cleaned-up.

So, instead of always applying the constraint for these types (including decimal), better check `IsPrecisionDefined` and `IsScaleDefined` on `col` and do not apply a constraint if they are not defined.

Reply:
Will try this approach soon.